src/Uploadcare/Uploader.php

```diff
@@ -158,29 +158,39 @@ private function fromUrlNew($url, $options = array())
      *
      * @param string $path
      * @param string|bool $mime_type
+     * @param string $filename
      * @return File
      * @throws \Exception
      */
-    public function fromPath($path, $mime_type = false)
+    public function fromPath($path, $mime_type = false, $filename = '')
     {
         if (function_exists('curl_file_create')) {
             if ($mime_type) {
-                $f = curl_file_create($path, $mime_type);
+                $f = curl_file_create($path, $mime_type, $filename);
             } else {
-                $f = curl_file_create($path);
+                $f = curl_file_create($path, null, $filename);
             }
         } else {
             if ($mime_type) {
                 $f = '@' . $path . ';type=' . $mime_type;
             } else {
                 $f = '@' . $path;
             }
+
+            if($filename) {
+              $f.=';filename='.$filename;
+            }
         }
 
         $data = array(
             'UPLOADCARE_PUB_KEY' => $this->api->getPublicKey(),
             'file' => $f,
         );
+
+        if ($filename != '') {
+            $data['filename'] = $filename;
```

Comment:
this will be ignored, as this endpoint doesn't expect filename as separate field in request data.

Reply:
Removed this.